### PR TITLE
Add GithubUtils and generateVersionComparisonURL

### PR DESCRIPTION
**Pull request by SombreroElGringo**
<If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>

### Details
I added the class `GithubUtils` and its method `generateVersionComparisonURL`.

PS: Also for the tests I needed to use `async/await` else I got some timeout. Also I tried different workaround for example `expect(Promise).resolves.toBe(url)` but it was not working correctly. At least with `async/await` we are sure the tests are working correctly.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes https://github.com/Expensify/Expensify.cash/issues/1577

### Tests
For repository with the following tags   `2.2.2`, `2.2.1`, `2.1.0-1`, `2.1.0`, `2.0.0-1`, `2.0.0`, `1.2.2-2`, `1.2.2-1`, `1.2.2`, `1.2.1`, `1.2.0`, `1.1.0`, `1.0.4`, `1.0.3`, `1.0.2`, `1.0.1`, `1.0.0` and `0.0.1`

- MAJOR comparison
-- if we pass the tag `2.2.2` as param it should return `https://<ETC>/1.2.2...2.2.2`
-- if we pass the tag `1.2.0` as param it should return `https://<ETC>/0.0.1...1.2.0`

- MINOR comparison
-- if we pass the tag `2.2.2` as param it should return `https://<ETC>/2.1.0...2.2.2`
-- if we pass the tag `1.1.0` as param it should return `https://<ETC>/1.0.4...1.1.0`

- PATCH comparison
-- if we pass the tag `2.2.2` as param it should return `https://<ETC>/2.2.1...2.2.2`
-- if we pass the tag `1.0.0` as param it should return `https://<ETC>/0.0.1...1.0.0`

- BUILD comparison
-- if we pass the tag `2.1.0-1` as param it should return `https://<ETC>/2.1.0...2.1.0-1`
-- if we pass the tag `1.2.2-2` as param it should return `https://<ETC>/1.2.2...1.2.2-2`

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

**Comment by SombreroElGringo** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by SombreroElGringo** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by roryabraham** [comment]
Hi @SombreroElGringo. Unfortunately, you've created this pull request in the wrong repo. As stated in the issue description, we need this code in https://github.com/Expensify/expensify-common instead of https://github.com/Expensify/Expensify.cash. (no worries, it was easy to miss in the description).

I'll review this PR to give you some feedback in the meantime, but then I'll be closing this PR.

File: .github/libs/GithubUtils.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+class GithubUtils {
+    constructor(octokit) {
+        this.octokit = octokit;
+    }
+
+    /**
+     * Generate a comparison URL between two versions following the semverLevel passed
+     *
+     * @param {String} repoSlug is the slug of the repository <owner>/<repository_name>
```
Comment (roryabraham) [inline .github/libs/GithubUtils.js:13]:
```suggestion
     * @param {String} repoSlug - The slug of the repository: <owner>/<repository_name>
```

File: .github/libs/GithubUtils.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+     * @param {String} tag is the tag to compare first the previous semverLevel
```
Comment (roryabraham) [inline .github/libs/GithubUtils.js:14]:
```suggestion
     * @param {String} tag - The tag to compare first the previous semverLevel
```

File: .github/libs/GithubUtils.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+     * @param {String} semverLevel is the semantic versioning MAJOR, MINOR, PATCH and BUILD
+     * @return {Promise} the url generated
+     */
+    generateVersionComparisonURL(repoSlug, tag, semverLevel) {
+        return new Promise((resolve, reject) => {
+            // eslint-disable-next-line max-len
```
Comment (roryabraham) [inline .github/libs/GithubUtils.js:20]: I would prefer that you don't disable eslint in this case.

File: .github/libs/GithubUtils.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            const getComparisonURL = (previousTag, currentTag) => `https://github.com/Expensify/Expensify.cash/compare/${previousTag}...${currentTag}`;
```
Comment (roryabraham) [inline .github/libs/GithubUtils.js:21]:
```suggestion
            const getComparisonURL = (previousTag, currentTag) => (
                `https://github.com/Expensify/Expensify.cash/compare/${previousTag}...${currentTag}`
            );
```

File: .github/libs/GithubUtils.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+            const [repoOwner, repoName] = repoSlug.split('/');
+            const tagSemver = semverParse(tag);
+
+            return this.octokit.repos.listTags({
+                owner: repoOwner,
+                repo: repoName,
+            })
+                .then(githubResponse => githubResponse.data.some(({name: repoTag}) => {
+                    if (
```
Comment (roryabraham) [inline .github/libs/GithubUtils.js:31]: The style we typically use here looks like:

```javascript
if (a
    && b
    || c
) {
    // Do something
}
```

And I'd prefer if you kept to that convention.

**Review by roryabraham — CHANGES_REQUESTED** [review]

**Pull request closed without merging**